src/middlewared/middlewared/plugins/network.py

```diff
@@ -818,6 +821,108 @@ def ipv4gw_reachable(self, ipv4_gateway):
         return False
 
 
+class StaticRouteService(CRUDService):
+    class Config:
+        datastore = 'network.staticroute'
+        datastore_prefix = 'sr_'
+        datastore_extend = 'staticroute.upper'
+
+    @accepts(Dict(
+        'staticroute_create',
+        IPAddr('destination'),
+        IPAddr('gateway'),
+        Str('description'),
+        register=True
+    ))
+    async def do_create(self, data):
+        await self.clean(data, 'staticroute_create')
+        await self.lower(data)
+
+        data['id'] = await self.middleware.call(
+            'datastore.insert', self._config.datastore, data,
+            {'prefix': self._config.datastore_prefix})
+
+        await self.middleware.call('service.start', 'routing')
+        await self.upper(data)
+
+        return data
+
+    @accepts(
+        Int('id'),
+        Patch(
+            'staticroute_create',
+            'staticroute_update',
+            ('attr', {'update': True})
+        )
+    )
+    async def do_update(self, id, data):
+        old = await self.middleware.call(
+            'datastore.query', self._config.datastore, [('id', '=', id)],
+            {'extend': self._config.datastore_extend,
+             'prefix': self._config.datastore_prefix,
+             'get': True})
+
+        new = old.copy()
+        new.update(data)
+
+        await self.clean(data, 'staticroute_update')
+
+        await self.lower(data)
+
+        await self.middleware.call(
+            'datastore.update', self._config.datastore, id, data,
+            {'prefix': self._config.datastore_prefix})
+
+        await self.middleware.call('service.start', 'routing')
+        await self.upper(new)
+
+        return new
+
+    @accepts(Int('id'))
+    async def do_delete(self, id):
+        return await self.middleware.call(
+            'datastore.delete', self._config.datastore, id)
+
+    @private
+    async def lower(self, data):
+        data['description'] = data['description'].lower()
+
+        return data
+
+    @private
+    async def upper(self, data):
+        data['description'] = data['description'].upper()
+
+        return data
+
+    @private
+    async def clean(self, staticroute, schema_name):
+        verrors = ValidationErrors()
+        dest = staticroute.get('destination')
+        dest = re.sub(r'\s{2,}', ' ', dest).strip()
+        raise_err = False
+
+        if dest.find('/') != -1:
+            try:
+                subnet_mask = int(dest.split('/')[1])
```

Comment:
What about `dest.split('/')[2]`? It may be present but `dest` will still be considered valid

Reply:
Example?

Reply:
`192.168.0.0/24/garbage` would be accepted as valid mask here